packages/brokeneck-react/src/components/Admin.js

```diff
@@ -44,13 +43,23 @@ export default function Admin() {
               <Redirect from="/:url*(/+)" to={pathname.slice(0, -1)} />
               <Redirect from="/" exact to="/users" />
               <Route path="/users" exact>
-                <Users />
+                <Entities
+                  entityName="User"
+                  entitiesName="Users"
+                  entitiesKey="users"
+                  query={LOAD_USERS}
+                />
```

Comment:
I think this might be part of the issue you've mentioned. Let's not move the abstraction at this level. Let's still have a `Users` and `Groups` components and let's make those components simply render `<Entities>` with the right props. It's a small redundancy but I think it's more correct that way, rather than moving knowledge about entity names and queries in this component here.

If you still see the issue you mentioned, let's discuss. I believe the issue will go away once you do this change, and it may be due to one or more of the hooks not implemented in the right way. Done this way you've done here, the `Entities` component will rerender with different props when navigating, which may cause the issue. If instead you wrap it in a Users and Groups components it will unmount instead. Or at least that's what I believe might be happening.